src/main/scala/com/ibm/developer/eventstore/akka/OnlineRetailWebServer.scala

```diff
@@ -19,62 +19,73 @@ import java.nio.charset.StandardCharsets
 import java.sql.Timestamp
 
 import akka.Done
-import akka.actor.{ ActorSystem, Terminated }
-import akka.event.Logging
-import akka.util.ByteString
-import akka.http.scaladsl.model.ws.{ BinaryMessage, Message, TextMessage }
-import akka.http.scaladsl.model.{ ContentTypes, HttpEntity }
-import akka.http.scaladsl.server.{ HttpApp, Route }
-import akka.http.scaladsl.settings.ServerSettings
-import akka.stream.{ ActorMaterializer, ActorMaterializerSettings, Supervision }
-import akka.stream.alpakka.csv.scaladsl.{ CsvParsing, CsvToMap }
+import akka.http.scaladsl.model.ContentTypes
+import akka.http.scaladsl.model.HttpEntity
+import akka.http.scaladsl.model.ws.BinaryMessage
+import akka.http.scaladsl.model.ws.Message
+import akka.http.scaladsl.model.ws.TextMessage
+import akka.http.scaladsl.server.HttpApp
+import akka.http.scaladsl.server.Route
+import akka.stream.Materializer
+import akka.stream.alpakka.csv.scaladsl.CsvParsing
+import akka.stream.alpakka.csv.scaladsl.CsvToMap
 import akka.stream.alpakka.ibm.eventstore.scaladsl.EventStoreSink
-import akka.stream.scaladsl.{ Flow, Keep, Source }
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.Sink
+import akka.util.ByteString
 import com.ibm.event.common.ConfigurationReader
 import org.apache.spark.sql.Row
 
-import scala.concurrent._
-import scala.util.Success
-
-class OnlineRetailWebServer extends HttpApp {
-
-  val decider: Supervision.Decider = { e =>
-    println("Unhandled exception in stream", e)
-    Supervision.Resume
-  }
```

Comment:
Nothing in the streams makes sure to resume on failure as far as I can see, but I could be wrong. The only operator used that adheres to the strategy is `.map` which means the map operations would continue feeding invalid data into the CSV parser after a failure. It could make sense for the `toRow` if you want to ignore entres that cannot be turned into rows - that is probably better modelled as a `Try` inside the stream instead if that is what you want.

Reply:
Additionally, setting this as default for the materializer would be problematic if the materializer is used to materialize the HTTP server blueprint where failures should fail a connection.